### ISSUE TITLE
Schedule.andThen repeats first element

### ISSUE THREAD

**Issue by reibitto** [zio/zio#3943]
I notice strange behavior with Schedule's `andThen`. These results are unexpected:

```scala
val schedule = Schedule.once.as(0.millis) andThen (Schedule.recurs(5) *> Schedule.exponential(1.second))

ZStream.fromSchedule(schedule.noDelay).foreach { x =>
  putStrLn(x.render)
}
```

prints out:

```
0 ns
0 ns
2 s
4 s
8 s
16 s
32 s
```

I see the same behavior when using `Schedule.run` rather than `ZStream.fromSchedule` which is why I'm thinking `Schedule.andThen` itself is the cause.

**Comment by reibitto** [comment]
Actually `Schedule.once` is `recurs(1)` which is why I think it's running twice. `Schedule.stop` is the one that is `recurs(0)`. Still though, the "1 s" being swallowed seems to indicate something is still wrong somewhere.

**Comment by adamgfraser** [comment]
Looks like #3976 fixes this issue:

```
0 ns
1 s
2 s
4 s
8 s
16 s
```

**Comment by adamgfraser** [comment]
Closed by #3976.
